src/coffee/extend/ForceDirectedGraph.coffee

```diff
@@ -0,0 +1,154 @@
+Function::property = (prop, desc) ->
+  Object.defineProperty @prototype, prop, desc
+
+
+# Hacky Edge implementation
+class Edge
+  constructor: (source, target) ->
+    @source = source
+    @target = target
+
+  @property 'p1',
+    get: ->
+      return {
+        x: @target.x,
+        y: @target.y,
+        z: @target.z
+      }
+
+  @property 'x',
+    get: ->
+      return @source.x
+
+  @property 'y',
+    get: ->
+      return @source.y
+
+  @property 'z',
+    get: ->
+      return @source.z
+
+  clone: () ->
+    return new Edge(@source.clone(), @target.clone())
+
+
+class Vertex extends Vector
+  constructor: () ->
+    super
+    @displacement = new Vector()
+
+class Graph
+  constructor: () ->
+    @vertices = []
+    @edges = []
+
+  addVertex: (vertex) ->
+    @vertices.push(vertex)
+    return @
+
+  addEdge: (edge) ->
+    @edges.push(edge)
+    return @
+
+
+class ForceDirectedGraph extends Graph
+  # Fruchterman & Reingold algorithm: http://citeseer.ist.psu.edu/viewdoc/download?doi=10.1.1.13.8444&rep=rep1&type=pdf
+  # Walshaw's algorithm: http://jgaa.info/accepted/2003/Walshaw2003.7.3.pdf
+
+  # ## Create a new force-directed graph. Vertices and Edges are stored in `this.vertices` and `this.edges`, respectively.
+  # @return new ForceDirectedGraph object
+  constructor: (width, height) ->
+    super()
+    @width = width
+    @height = height
+    @frames = []
+
+  _fa: (x, k) ->
```

Comment:
May want to move `_fa`, `_fa`, and `_cool` inside `generate`, since they are only used inside it.